cms/djangoapps/contentstore/views/course.py

```diff
@@ -469,6 +440,52 @@ def course_index(request, course_key):
         })
 
 
+def get_courses_accessible_to_user(request):
+    """
+    Try to get all courses by first reversing django groups and fallback to old method if it fails
+    Note: overhead of pymongo reads will increase if getting courses from django groups fails
+    """
+    if GlobalStaff().has_user(request.user):
+        # user has global access so no need to get courses from django groups
+        courses, in_process_course_actions = _accessible_courses_list(request)
+    else:
+        try:
+            courses, in_process_course_actions = _accessible_courses_list_from_groups(request)
+        except AccessListFallback:
+            # user have some old groups or there was some error getting courses from django groups
+            # so fallback to iterating through all courses
+            courses, in_process_course_actions = _accessible_courses_list(request)
+    return courses, in_process_course_actions
+
+
+def _remove_in_process_courses(courses, in_process_course_actions):
+    """
+    remove any courses in courses that are also in the in_process_course_actions list and formats them for view
```

Comment:
Nit: I know you just refactored this code, but it would be helpful for you to add to the documentation what an "in_process_course" is. Both times I read this PR I got confused because I thought those are the courses the user is in the process of taking. But looking at _accessible_courses_list_from_groups, I see that "in-process" actually refers to courses that are in the process of being generated for re-run.